File: umu/src/main/java/com/wyy/service/UmmClient.java
package com.wyy.service;

import com.alibaba.fastjson.JSONObject;
import com.wyy.client.AuthorizedFeignClient;
import com.wyy.domain.*;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@AuthorizedFeignClient(name = "umm")
public interface UmmClient {

    @RequestMapping(value ="/api/tasks", method = RequestMethod.GET)
    List<Task> getTasks(@RequestParam(value = "uuid") String uuid);

    @RequestMapping(value ="/api/tasks", method = RequestMethod.POST)
    ResponseEntity<Task> createTask(@RequestBody Task task);

    @RequestMapping(value ="/api/tasks", method = RequestMethod.PUT)
    ResponseEntity<Task> updateTask(@RequestBody Task task);

    @RequestMapping(value ="/api/task-steps", method = RequestMethod.POST)
    ResponseEntity<TaskStep> createTaskStep(@RequestBody TaskStep taskStep);

    @RequestMapping(value ="/api/solutions", method = RequestMethod.GET)
    List<Solution> getSolutions(@RequestParam(value = "uuid") String uuid);

    @RequestMapping(value ="/api/solutions", method = RequestMethod.POST)
    ResponseEntity<Void> createSolution(@RequestBody Solution solution);

    @RequestMapping(value ="/api/solutions/picture-url", method = RequestMethod.PUT)
    ResponseEntity<Solution> updateSolutionPictureUrl(@RequestBody JSONObject jsonObject);

    @RequestMapping(value ="/api/solutions/{id}", method = RequestMethod.DELETE)
    ResponseEntity<Void> deleteSolution(@PathVariable(value = "id") Long id);

    @RequestMapping(value ="/api/artifacts", method = RequestMethod.POST)
    ResponseEntity<Artifact> createArtifact(@RequestBody Artifact artifact);

    @RequestMapping(value ="/api/artifacts", method = RequestMethod.GET)
    List<Artifact> getArtifacts(@RequestParam(value = "solutionUuid") String solutionUuid, @RequestParam(value = "type") String type);

    @RequestMapping(value ="/api/artifacts", method = RequestMethod.GET)
    List<Artifact> getAllArtifacts(@RequestParam(value = "solutionUuid") String solutionUuid);

    @RequestMapping(value ="/api/artifacts/{id}", method = RequestMethod.DELETE)
    ResponseEntity<Void> deleteArtifact(@PathVariable(value = "id") Long id);

    @RequestMapping(value ="/api/documents", method = RequestMethod.POST)
    ResponseEntity<Document> createDocument(@RequestBody Document document);

    @RequestMapping(value ="/api/documents/{id}", method = RequestMethod.DELETE)
    ResponseEntity<Void> deleteDocument(@PathVariable(value = "id") Long id);

    @RequestMapping(value ="/api/documents/{id}", method = RequestMethod.GET)
    ResponseEntity<Document> getDocument(@PathVariable(value = "id") Long id);


}

File: umu-python/app/service/umm_client.py
from app.service.http_client import http_client


service_name = 'umm'


def create_task(task, jwt):
    body = {
        'action': 'create_task',
        'args': {
            'task': task.__dict__,
        }
    }
    return http_client(service_name, body=body, jwt=jwt)


def update_task(task, jwt):
    body = {
        'action': 'update_task',
        'args': {
            'task': task.__dict__,
        }
    }
    return http_client(service_name, body=body, jwt=jwt)


def create_task_step(task_step, jwt):
    body = {
        'action': 'create_task_step',
        'args': {
            'task_step': task_step.__dict__,
        }
    }
    return http_client(service_name, body=body, jwt=jwt)


def create_solution(solution, jwt):
    body = {
        'action': 'create_solution',
        'args': {
            'solution': solution.__dict__,
        }
    }
    return http_client(service_name, body=body, jwt=jwt)


def get_solutions(uuid, jwt=None):
    body = {
        'action': 'get_solutions',
        'args': {
            'uuid': uuid,
        }
    }
    return http_client(service_name, body=body, jwt=jwt)


def delete_solution(solutionId, jwt):
    body = {
        'action': 'delete_solution',
        'args': {
            'solutionId': solutionId,
        }
    }
    return http_client(service_name, body=body, jwt=jwt)


def create_artifact(artifact, jwt):
    body = {
        'action': 'create_artifact',
        'args': {
            'artifact': artifact.__dict__,
        }
    }
    return http_client(service_name, body=body, jwt=jwt)


def get_artifacts(solution_uuid, jwt=None):
    body = {
        'action': 'get_artifacts',
        'args': {
            'solutionUuid': solution_uuid,
        }
    }
    return http_client(service_name, body=body, jwt=jwt)


def delete_artifact(artifactId, jwt):
    body = {
        'action': 'delete_artifact',
        'args': {
            'artifactId': artifactId,
        }
    }
    return http_client(service_name, body=body, jwt=jwt)


def create_document(document, jwt):
    body = {
        'action': 'create_document',
        'args': {
            'document': document.__dict__,
        }
    }
    return http_client(service_name, body=body, jwt=jwt)


def find_document(documentId, jwt=None):
    body = {
        'action': 'find_document',
        'args': {
            'documentId': documentId,
        }
    }
    return http_client(service_name, body=body, jwt=jwt)


def delete_document(documentId, jwt):
    body = {
        'action': 'delete_document',
        'args': {
            'documentId': documentId,
        }
    }
    return http_client(service_name, body=body, jwt=jwt)
